Add condition for system setZone

davkub opened the pull request:
**What is this about:**
The original issue that led to creating this PR was `setZone('system')` is using the Timezone in `Settings.defaultZone`, not the browser Timezone (https://github.com/moment/luxon/issues/1182), the name is pretty self-explanatory.

**Changes in this PR:**

- added a condition for `system` to return `SystemZone.instance` along with `local`
- added a condition for `defaultZone` to accept also `default` parameter
- Fixed deprecated tests that took local zone as default while setting non-system one, as per "local" is a legacy name for "system"
- Add test for `setZone()` using the `system` zone

davkub commented:
/easycla

davkub commented:
/easycla

icambron commented on src/impl/zoneUtil.js:
```diff
@@ -17,7 +18,8 @@ export function normalizeZone(input, defaultZone) {
   } else if (isString(input)) {
     const lowered = input.toLowerCase();
-    if (lowered === "local" || lowered === "system") return defaultZone;
+    if (lowered === "local" || lowered === "default") return defaultZone;
```
"local" is a legacy name for "system", so local should have followed system over to the new condition

davkub replied:
@icambron This has been refactored, local condition moved to the system one.

icambron commented:
Try running tests with `scripts/test`, which set some env vars

davkub commented:
> Try running tests with `scripts/test`, which set some env vars
@icambron 
I have updated "deprecated" tests that used setZone('local') as a default one, currently local refers to the system zone as you described in your comment. Also added 1 more setZone test for system zone option.

icambron commented:
Ok, this is ready to go. I just need to decide if I can get away with releasing this as Luxon 2.5 or if it needs to be called 3.0

davkub commented:
> Ok, this is ready to go. I just need to decide if I can get away with releasing this as Luxon 2.5 or if it needs to be called 3.0

Hello, have you decided which version are we going with?

icambron commented:
@davkub I'm releasing this in 3.0 today
